src/lib.rs

```diff
@@ -265,25 +382,51 @@ impl<R: gimli::Reader> Context<R> {
                 if probe_low >= i.range.end || probe_high <= i.range.begin {
                     return None;
                 }
-                Some((&self.dwarf.units[i.unit_id], &i.range))
+                Some((&self.parsed_dwarf.units[i.unit_id], &i.range))
             })
     }
 
     /// Find the DWARF unit corresponding to the given virtual memory address.
-    pub fn find_dwarf_unit(&self, probe: u64) -> Option<&gimli::Unit<R>> {
-        for unit in self.find_units(probe) {
-            match unit.find_function_or_location(probe, &self.dwarf) {
-                Ok((Some(_), _)) | Ok((_, Some(_))) => return Some(&unit.dw_unit),
-                _ => {}
-            }
+    pub fn find_dwarf_and_unit(
+        &self,
+        probe: u64,
+    ) -> LookupResult<
+        impl LookupContinuation<Output = Option<(&gimli::Dwarf<R>, &gimli::Unit<R>)>, Buf = R>,
+    > {
+        let mut units_iter = self.find_units(probe);
+        if let Some(unit) = units_iter.next() {
```

Comment:
Can you make this code and the similar code in `find_frames` have the same style (either both use `if let` or both use `match`), so that it is clearer that they are doing similar things.